src/Psy/VersionUpdater/CheckerFactory.php

```diff
@@ -0,0 +1,24 @@
+<?php
+
+/*
+ * This file is part of Psy Shell.
+ *
+ * (c) 2012-2015 Justin Hileman
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Psy\VersionUpdater;
+
+use GuzzleHttp\Client;
+
+class CheckerFactory
```

Comment:
`Configuration` acts as our factory and dependency injection container, especially for simple things like this.